com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/archive/ArchiveBuilder.java

```diff
@@ -211,14 +212,17 @@ public void write(RandomAccessFile archiveIndex, RandomAccessFile archiveData, P
 
             // Calculate hash digest values for resource
             String hexDigest = null;
-            try {
-                byte[] hashDigest = ManifestBuilder.CryptographicOperations.hash(buffer, manifestBuilder.getResourceHashAlgorithm());
-                entry.setHash(new byte[HASH_MAX_LENGTH]);
-                System.arraycopy(hashDigest, 0, entry.getHash(), 0, hashDigest.length);
-                hexDigest = ManifestBuilder.CryptographicOperations.hexdigest(hashDigest);
-            } catch (NoSuchAlgorithmException exception) {
-                throw new IOException("Unable to create a Resource Pack, the hashing algorithm is not supported!");
+            byte[] hashDigest;
+            if (ResourceDigestCache.hasDigest(entry.getFilename())) {
```

Comment:
Use existing resource digest calculated at some earlier time (in our case while building the Lua source files)